crossfire/__init__.py

```diff
@@ -51,16 +61,42 @@ def occurrences(
     )
 
 
+def _flatten_df(row, column_name):
+    column_data = row[column_name]
+    return Series(
+        {f"{column_name}_{key}": value for key, value in column_data.items()}
+    )
+
+
+def is_empty(data):
+    if HAS_PANDAS and isinstance(data, DataFrame):
+        return data.empty
+    else:
+        return not data
+
+
 def flatten(data, nested_columns=None):
     nested_columns = set(nested_columns or NESTED_COLUMNS)
     if not nested_columns.issubset(NESTED_COLUMNS):
         raise NestedColumnError(nested_columns)
-    if not data:
+    if is_empty(data):
         return data
-    if isinstance(data, list):
-        keys = set(data[0].keys()) & nested_columns
-        for item in data:
-            for key in keys:
-                item.update({f"{key}_{k}": v for k, v in item.get(key).items()})
-                item.pop(key)
+    if HAS_PANDAS and isinstance(data, (DataFrame, GeoDataFrame)):
```

Comment:
Do we need `GeoDataFrame`?

```suggestion
    if HAS_PANDAS and isinstance(data, DataFrame):
```

Checking the documentation for `isinstance` I think we don't.